core/cap-0062.md

```diff
@@ -0,0 +1,281 @@
+version: number identifying the file format version
+server: an optional debugging string identifying the software that wrote the file
+networkPassphrase: an optional string identifying the networkPassphrase
+currentLedger: a number denoting the ledger this file describes the state of
+currentBuckets: an array containing an encoding of the live state bucket list for this ledger
+hotArchiveBuckets: an array containing an encoding of the hot archive bucket list for this ledger
+```
+
+Bucket files for the `hotArchiveBuckets` will be stored just as Bucket files are currently stored.
+
+#### Changes to LedgerHeader
+
+While the XDR structure of `LedgerHeader` remains unchanged, `bucketListHash` will be changed to the following:
+
+```
+header.bucketListHash = SHA256(liveStateBucketListHash, hotArchiveHash)
+```
+
+#### Meta
+
+Whenever a `PERSISTENT` entry is evicted (i.e. removed from the Live State BucketList and added to the Hot Archive),
+the entry key and its associated TTL key will be emitted via `evictedLedgerKeys`
+(this field has been renamed and was previously named `evictedTemporaryLedgerKeys`).
+
+Whenever an entry is restored via `RestoreFootprintOp`, the `LedgerEntry` being restored and its
+associated TTL will be emitted as a `LedgerEntryChange` of type `LEDGER_ENTRY_RESTORE`. Note that
+entries being restored may not have been evicted such that entries currently in the Live State BucketList
+can see `LEDGER_ENTRY_RESTORE` events.
+
+#### Default Values for Network Configs
+
+`sorobanLiveStateTargetSizeBytes` (formerly `bucketListTargetSizeBytes`) will need to be changed
+to reflect the new size calculations. Additionally, the "fee curve" may need to be revisted
+now that classic state growth no longer affects Soroban fees.
+
+## Design Rationale
+
+### CAP-0062 vs. CAP-0057
+
+This CAP is a strict subset of [CAP-0057](cap-0057.md). The intention is not to replace CAP-0057, but to implement this CAP
+first as a step towards CAP-0057. At current usage levels, "full state archival" as specified by CAP-0057 (i.e. evicted entries are deleted
+from validators) is not a necessary or useful optimization. The space savings of validators and the History Archive would be
+minimal, and the additional complexity, both from a computational and UX standpoint, do not justify the minimal storage savings.
+Long term, should usage increase, the tradeoffs introduced by CAP-0057 will make more sense, but that is not the case currently
+or in the near term future.
+
+The reasoning behind introducing CAP-0057 before it was strictly necessary was to prevent breakage of applications. If applications
+and users were not prepared for State Archival, it would be very difficult to enable CAP-0057. However, because
+[CAP-0046-12](./cap-0046-12.md) is implemented and State Archival semantics are already required, upgrading to CAP-0057 should not
+be breaking, especially since much of the complexity, like proof generation, is handled automatically by RPC endpoints.
```

Comment:
```suggestion
be breaking, especially since much of the complexity, like proof generation, will be handled automatically by new Stellar Core HTTP endpoints.
```

Is that correct? While RPC will be the facilitator via `simulateTransaction`, the proofs are actually generated by `/getrestorationproof` per https://github.com/stellar/go/issues/5426, right?

Reply:
Yeah, I meant from a contract invoker's perspective it's handled by RPC. I'll clarify.